### PR TITLE
Fix #16958: column parser data type fix

### LINKED ISSUE #16958: Superset ingestion fails due to data type bug
**Affected module**
Ingestion Framework - Superset

**Describe the bug**
Superset ingestion fails due to data type bug.
[SupersetDebugLogOM_1.4.3.txt](https://github.com/user-attachments/files/16127179/SupersetDebugLogOM_1.4.3.txt)

**To Reproduce**
1) create dashboard or chart in Superset connected with table including string column
2) create Superset connection in OpenMetadata
3) run metadata ingestion 

**Expected behavior**
Sucсessful metadata ingestion.

**Version:**
 - Ubuntu 22.04.3 LTS (GNU/Linux 5.15.0-102-generic x86_64)
 - Python version: Python 3.10.13
 - OpenMetadata version: 1.4.3
 - OpenMetadata Ingestion package version: openmetadata-ingestion==1.4.3.1

**Additional context**
- Slack thread: https://openmetadata.slack.com/archives/C02B6955S4S/p1720096397025029

### PR DESCRIPTION
**Pull request by harshsoni2024**
### Describe your changes:

Fixes #16958

I worked on ... because ...

#
### Type of change:

- [x] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/database/column_type_parser.py [hunk 1]
```diff
@@ -418,6 +418,9 @@ def _parse_struct_fields_string(stuct_type: str) -> Dict[str, object]:
     def _parse_primitive_datatype_string(  # pylint: disable=too-many-return-statements
         dtype: str,
     ) -> Dict[str, object]:
+        empty_brackets = "()"
+        if empty_brackets in dtype:
+            dtype = dtype.replace(empty_brackets, "")
```
Comment (ulixius9) [inline ingestion/src/metadata/ingestion/source/database/column_type_parser.py:423]: this should be in superset/metadata

File: ingestion/src/metadata/ingestion/source/dashboard/superset/mixin.py [hunk 2]
```diff
@@ -223,6 +223,12 @@ def _get_datamodel(
             )
         return None
 
+    def _clearn_column_datatype(self, datatype: str) -> str:
+        """clean datatype of column fetched from superset"""
+        if "()" in datatype:
+            datatype = datatype.replace("()", "")
```
Comment (ulixius9) [inline ingestion/src/metadata/ingestion/source/dashboard/superset/mixin.py:229]:
```suggestion
    datatype = datatype.replace("()", "")
```
```suggestion
        if "()" in datatype:
            datatype = datatype.replace("()", "")
```

**Comment by harshach** [comment]
@ulixius9 ok to merge?

**Comment by harshsoni2024** [comment]
@harshach, I think it's ok to merge now. previously failing checks are passed now !

**Review by ulixius9 — APPROVED** [review]

**Pull request merged**